modules/flowable-engine-common/src/main/java/org/flowable/common/engine/impl/AbstractEngineConfiguration.java

```diff
@@ -502,7 +502,19 @@ public void initDatabaseType() {
 
             // CRDB does not expose the version through the jdbc driver, so we need to fetch it through version().
             if (PRODUCT_NAME_POSTGRES.equalsIgnoreCase(databaseProductName)) {
```

Comment:
The logic is in this if - i.e. it will only be triggered when databaseProductName matches postgres. 

This value is determined above:
  DatabaseMetaData databaseMetaData = connection.getMetaData();
  String databaseProductName = databaseMetaData.getDatabaseProductName();

For oracle, db2, etc. the code in the if should thus never be reached. 
How did you reproduce bug on those databases?

Reply:
I'm not sure I follow that reasoning, the code is already generic for each database:

databaseType = databaseTypeMappings.getProperty(databaseProductName);

which is based on the database metadata. Only for CRDB/Postgres an extra check is done, because CRDB returns the same value in its metadata as postgres.

Reply:
@pandaapo I am not sure which error you are trying to fix. The error in #3400 is within Liquibase not within Flowable